lib/email/styles.rb

```diff
@@ -244,7 +246,8 @@ def inline_secure_images(attachments)
       stripped_media.each do |div|
         url = div['data-stripped-secure-media']
         filename = File.basename(url)
-        sha1 = filename.gsub(File.extname(filename), "")
+        filename_bare = filename.gsub(File.extname(filename), "")
+        sha1 = filename_bare.include?('_') ? filename_bare.partition('_').first : filename_bare
```

Comment:
I think it would be a little bit easier to read without that check 
```
 sha1 =  filename_bare.partition('_').first # that should work for filenames without _ as well
```